SampleAPI/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using SampleAPI.Constant;
+using SampleAPI.Models;
+using SampleAPI.Repository;
+
+namespace SampleAPI.Controllers
+{
+    /// <summary>
+    /// EmployeeModel controller is used to perform CRUD operations with employee entity.
+    /// </summary>
+    [Route("sampleAPI/[controller]")]
+    [ApiController]
+    public class EmployeeController : ControllerBase
+    {
+        private readonly ILogger<EmployeeController> _logger;
+        private readonly IEmployeeRepository _repository;
+        public EmployeeController(ILogger<EmployeeController> logger, IEmployeeRepository repository)
+        {
+            _logger = logger;
+            _repository = repository;
+        }
+
+        /// <summary>
+        /// Get all employee records.
+        /// </summary>
+        /// <returns>Returns the employee list.</returns>
+        [HttpGet]
+        [Route("GetAllEmployee")]
+        public async Task<IActionResult> GetEmployee(string? search, string? filterByRole, bool includeInActive, [FromQuery] PagingParameterModel pagingparametermodel)
```

Comment:
you have marked [FromQuery] on just the paginationParameterModel, but all the parameters to this action will be mapped from query parameters because it is a GET. 

Like you have created a model for just the pagination, you can actually wrap all the other parameters into a class and use that here ? Something like GetEmployeeParameters ?

Reply:
I mean GetEmployeeParameters would have all the parameters including the PageNumber and PageSize ones